Add meta.pkgConfigModules to gtk4 and dependencies

rvl opened the pull request:
## Description of changes

This is PR #285125 from @eldritch-cookie, with review comments fixed, and rebased.

## Things done

Built on platform: `x86_64-linux`

I used the following file to test the changes:
```nix
{ pkgs ? import ./default.nix { } }:

with pkgs;

let
  changed = [
    appstream
    expat
    fontconfig
    fribidi
    graphene
    gtk4
    isocodes
    libadwaita
    libdeflate
    libepoxy
    libffi
    libtiff
    libxkbcommon
    libxml2
    pango
    pixman
    graphite2
    tracker
    vulkan-loader
    wayland
    xz
  ];

  relevant = [
    libxml2Python
  ];

in
{
  together = linkFarmFromDrvs "eval-meta-pkgConfigModules" (changed ++ relevant);

  newChecks = linkFarmFromDrvs "test-pkgConfigModules"
    (lib.concatMap (drv: map (output: output.tests.pkg-config) drv.all) changed);

  metas = let
    dumpJSON = xs: "[\n${lib.concatMapStringsSep ",\n" builtins.toJSON xs}\n]\n";
    getMeta = drv: builtins.removeAttrs drv.meta [ "pkgConfigModules" "position" ];
  in
    runCommandLocal "metas.json" {
    nativeBuildInputs = [ jq ];
    metas = dumpJSON (map getMeta changed);
  } ''
    jq . <<< "$metas" > $out
  '';
}
```

To ensure that only package `meta` has changed, not the derivations, I compare store paths before and after applying these changes:

```
$ nix path-info -f ./test-meta-pkgConfigModules.nix together
/nix/store/q072z4gx5wa16lw15rjh7r4cy867vki7-eval-meta-pkgConfigModules
```

And to run `testers.hasPkgConfigModules` for each modified package:
```
$ nix build -L -f test-meta-pkgConfigModules.nix newChecks
```

Checking that `meta` has not changed for any package (besides `meta.pkgConfigModules`), I run this before and after applying these changes, and check that the resulting file is exactly the same:
```
$ nix build -L -f test-meta-pkgConfigModules.nix metas -o metas-after
```

eldritch-cookie committed:
gtk4: add meta.pkgConfigModules and tests.pkg-config

Co-authored-by: Rodney Lorrimar <dev@rodney.id.au>

eldritch-cookie committed:
graphene: add pkgConfigModules metadata and corresponding test

eldritch-cookie committed:
isocodes: add pkgConfigModules metadata and corresponding test

eldritch-cookie committed:
pango: add tests for pkg-config modules

eldritch-cookie committed:
fribidi: add pkgConfigModules metadata and corresponding test

eldritch-cookie committed:
libepoxy: added test for pkg-config modules

eldritch-cookie committed:
libxkbcommon: add tests for pkg-config modules

eldritch-cookie committed:
libtiff: add pkgConfigModules metadata and corresponding test

do note that while the test added ran and passed there were failing
tests.

eldritch-cookie committed:
libxml2: add meta.pkgConfigModules and tests.pkg-config

Co-authored-by: Rodney Lorrimar <dev@rodney.id.au>

eldritch-cookie committed:
tracker: add meta.pkgConfigModules and tests.pkg-config

eldritch-cookie committed:
vulkan-loader: add meta.pkgConfigModules

Co-authored-by: Rodney Lorrimar <dev@rodney.id.au>

rvl committed:
expat: Add meta.pkgConfigModules

rvl committed:
fontconfig: Add meta.pkgConfigModules

rvl committed:
libdeflate: Add meta.pkgConfigModules

rvl committed:
xz: Add meta.pkgConfigModules

rvl committed:
libffi: Add meta.pkgConfigModules

rvl committed:
pixman: Add meta.pkgConfigModules

rvl committed:
graphite2: Add meta.pkgConfigModules

rvl committed:
wayland: Add meta.pkgConfigModules and corresponding test

rvl committed:
appstream: Add meta.pkgConfigModules

rvl committed:
libadwaita: Add meta.pkgConfigModules and tests.pkg-config

eldritch-cookie reviewed (changes requested):
from reading the ofborg automated checks on the meta.changelog attribute there is a reference to src which given that rec was removed should refer to finalAttrs.src https://github.com/NixOS/nixpkgs/blob/1454e6ff9a743ac6df84e810e060f7c25c67038b/pkgs/development/libraries/libadwaita/default.nix#L118

rvl commented:
> from reading the ofborg automated checks on the meta.changelog attribute there is a reference to src which given that rec was removed should refer to finalAttrs.src

Thanks, fixed.

rvl commented on pkgs/development/libraries/libxml2/default.nix:
```diff
@@ -32,7 +33,7 @@ in
 
 let
-libxml = stdenv.mkDerivation rec {
+libxml = stdenv.mkDerivation (finalAttrs: rec {
```
This `rec` I have kept because libxml package does something with `passthru` which I don't understand, and I don't want to break anything.

rvl commented:
I fixed a rather grievous error in vulkan-loader, rebased on latest `nixos-unstable`, and did some more testing to check the integrity of existing `meta` attributes.
